### DISCUSSION TITLE
Reasons why bromite was rejected by the play store

### DISCUSSION THREAD

**Discussion by uazo** [bromite/bromite#1987] [General]
I apologize in advance for the matter, mine is pure curiosity, but based on [this](https://github.com/bromite/bromite/blob/master/FAQ.md#is-bromite-on-play-store) it seems that bromite is not being accepted (also) due to the adblock.

I am checking the implementation of the adblocks in the other chromium forks, sincerely not finding substantial differences with the logics adopted by bromite compared to the others, I was expecting some while lists but I didn't find them, but perhaps because I am still at the beginning of understanding 

among other things I still don't understand why everyone says that bromite adblock is not as good as the others, if we mean by adblock an effective blocking of the ad's http request, rather than its display.

if it's not too complex (and it doesn't take too long), is it possible to have some information (or links) on the reasons why bromite was not accepted due the adblock?
thank you very much (obviously without any rush)

@csagan5 in any case, it is not to then propose it, but it is only to understand, also because reading [this](https://support.google.com/googleplay/android-developer/answer/11498144) I do not understand which voice bromite does not follow

**Comment by csagan5** [comment]
> is it possible to have some information (or links) on the reasons why bromite was not accepted due the adblock?

There are old conversations around but basically:
* other Chromium forks have been asked to not ad-block certain Google-owned URLs (Kiwi author said so, I do not know about others)
* when I made the submission I was told by the semi-automated Play Store answers that the app was trying to look like another existing app

**Reply by uazo** [reply]
a thousand thanks. my question was only related to the adblock, simply because I not finding anything abnormal and I wondered why

I'll let you know if I find anything useful for our adblock in the code of the other forks (and the way to exploit it), then let's see if it can be useful to integrate it.

**Reply by csagan5** [reply]
> then let's see if it can be useful to integrate it.

Useful for what purpose?

**Reply by uazo** [reply]
> Useful for what purpose?

to improve the adblock

basically, at the moment, I saw:

- it is possible to acquire more rules via a simple modification in rule_parser, adding the new aliases defined [here](https://github.com/gorhill/uBlock/wiki/Static-filter-syntax) in  [rule_options.h](https://source.chromium.org/chromium/chromium/src/+/main:components/subresource_filter/tools/rule_parser/rule_options.h;l=62;bpv=1;bpt=1?q=%7Burl_pattern_index::proto::ELEMENT_TYPE_SUBDOCUMENT&ss=chromium%2Fchromium%2Fsrc)
- rule_parser already manages the acquisition of the rules relating to the [CSS](https://source.chromium.org/chromium/chromium/src/+/main:components/subresource_filter/tools/rule_parser/rule_parser.cc;l=415;bpv=1;bpt=1) but that transformation in indexed rules [ignores](https://source.chromium.org/chromium/chromium/src/+/main:components/subresource_filter/core/common/unindexed_ruleset.h;l=83?q=UnindexedRulesetWriter&ss=chromium%2Fchromium%2Fsrc) them
- the code needed to manage some rules is visible in [vivaldi](https://github.com/ric2b/Vivaldi-browser/tree/master/components/request_filter/adblock_filter), including cosmetic filters

the process seems structured as follows:
- the parser switches from the textual rules to the [proto format](https://source.chromium.org/chromium/chromium/src/+/main:components/url_pattern_index/proto/rules.proto)
- the indexing phase produces a [flat](https://source.chromium.org/chromium/chromium/src/+/main:components/url_pattern_index/flat/url_pattern_index.fbs) file
- that the entry point of the adblock check in chromium is [UrlPatternIndexMatcher](https://source.chromium.org/chromium/chromium/src/+/main:components/url_pattern_index/url_pattern_index.cc;l=894?q=UrlPatternIndexMatcher::FindMatch&sq=&ss=chromium%2Fchromium%2Fsrc)

for now I have arrived here

**Reply by csagan5** [reply]
> > Useful for what purpose?
> 
> to improve the adblock

There is no intention to abide to adblocker restrictions of the Play Store since Bromite has never been there.

> * the code needed to manage some rules is visible in [vivaldi](https://github.com/ric2b/Vivaldi-browser/tree/master/components/request_filter/adblock_filter), including cosmetic filters

Does Vivaldi still use the subresource filter?

**Reply by uazo** [reply]
> Does Vivaldi still use the subresource filter?

I don't know, but honestly I only care how (and where) it injects the css generated by the filters and compare it with brave and [this](https://github.com/seclab-ucr/ShadowBlock/blob/master/patches/patch.diff).
or figure out how to use [libadblockplus](https://github.com/adblockplus/libadblockplus) that I found in the "sources" of edge

**Reply by csagan5** [reply]
Some new functions could be called in places where the extension hooks are normally called (in Manifest V2 or V3).

**Reply by uazo** [reply]
> where the extension hooks are normally called

too complex to follow, I prefer to check differently, anyway thanks for the indication.
